res/sections/80-AndroidExamples.tex

```diff
@@ -455,14 +455,313 @@ \section{Lab 4}
+mMap.setOnMarkerClickListener(this);
+
+mMap.moveCamera(CameraUpdateFactory
+					.newLatLngZoom(PRATO_VALLE_ADDRESS, 15));
+mMap.animateCamera(CameraUpdateFactory.zoomTo(10), 2000, null);
+\end{lstlisting}
+
+Si mostra ora l'implementazione dei metodi dei listener.
+
+\begin{lstlisting}[language=Java]
+@Override
+public void onMapClick(LatLng point) {
+	mMap.animateCamera(CameraUpdateFactory.newLatLng(point));   
+	mMarkerIsClicked = false;
+}
+
+@Override
+public void onMapLongClick(LatLng point) {
+	mMap.addMarker(new MarkerOptions()
+						.position(point)
+						.title(point.toString()));    
+	mMarkerIsClicked = false;
+}
+
+@Override
+public boolean onMarkerClick(Marker marker) {
+		if (mPolyline != null) {
+		mPolyline.remove();
+		mPolyline = null;
+	}
+
+	if (mMarkerIsClicked) {
+		// disegna e mostra nella mappa una linea rossa 
+		// che parte dal precedente marker e finisce nell'ultimo marker
+		mRectOptions.add(marker.getPosition());
+		mRectOptions.color(Color.RED);
+		mPolyline = mMap.addPolyline(mRectOptions);
+	} else {
+		// Nuovo marker, salvo la sua posizione
+		mRectOptions = new PolylineOptions()
+							.add(marker.getPosition());
+		mMarkerIsClicked = true;
+	}
+	return true;
+}
+\end{lstlisting}
+
+
 \subsection{NotifyDemo}
 
-\subsection{BindDemo}
+L'applicazione mostra l'implementazione di un \lstinline|Service| e la comunicazione di esso con un'\lstinline|Activity|, nel codice chiamati \lstinline|NotifyService| e \lstinline|NotifyActivity| rispettivamente. Il \lstinline|Service| lancerà una notifica mentre dall'\lstinline|Activity| sarà possibile farlo partire e/o fermarlo tramite il meccanismo di \lstinline|Intent| e \lstinline|BroadcastReceiver|.
+
+Dichiariamolo nel manifest:
+\begin{lstlisting}[language=Java]
+<application>
+	<service android:name="math.unipd.it.mp.notifydemo.NotifyService"/>
+</application>
+\end{lstlisting}
+
+Nel metodo \lstinline|onCreate()| dell'\lstinline|Activity| settiamo due listener ai pulsanti per avviare e fermare il \lstinline|Service|.
+\begin{lstlisting}[language=Java]
+Button buttonStartService = 
+	(Button)findViewById(R.id.startservice);
+Button buttonStopService = 
+	(Button)findViewById(R.id.stopservice);
+
+buttonStartService.setOnClickListener(
+	new Button.OnClickListener(){
+		public void onClick(View arg0) {
+			Intent intent = 
+				new Intent(NotifyActivity.this, NotifyService.class);
+			startService(intent);
+		}
+	});
+
+buttonStopService.setOnClickListener(
+	new Button.OnClickListener(){
+		public void onClick(View arg0) {
+			Intent intent = new Intent();
+			intent.setAction(NotifyService.ACTION);
+			intent.putExtra(
+				NotifyService.STOP_SERVICE_BROADCAST_KEY, 
+				NotifyService.RQS_STOP_SERVICE);
+			sendBroadcast(intent);
+		}
+	}
+);
+\end{lstlisting}
+
+Di seguito una implementazioe di un \lstinline|BroadcastReceiver| che serve per comunicare con l'\lstinline|Service| dall'\lstinline|Activity| (vale anche il contrario).
```

Comment:
`implementazioe` da correggere in `implementazione`. Poi `una` forse e' meglio `un'`?